skimage/feature/tests/test_blob.py

```diff
@@ -142,32 +142,42 @@ def test_blob_dog_3d_anisotropic(dtype, threshold_type):
     assert abs(math.sqrt(3) * b[5] - r) < 1.1
 
 
-def test_blob_dog_excl_border():
+@pytest.mark.parametrize("disc_center", [(5,5), (5, 20)])
+@pytest.mark.parametrize("exclude_border", [6, (6, 6), (4,15)])
+def test_blob_dog_exclude_border(disc_center, exclude_border):
     # Testing exclude border
 
-    # image where blob is 5 px from borders, radius 5
+    # image where blob is disc_center px from borders, radius 5
     img = np.ones((512, 512))
-    xs, ys = disk((5, 5), 5)
+    xs, ys = disk(disc_center, 5)
     img[xs, ys] = 255
     blobs = blob_dog(
         img,
         min_sigma=1.5,
         max_sigma=5,
         sigma_ratio=1.2,
     )
-    assert blobs.shape[0] == 1
+    assert blobs.shape[0] == 1, "one blob should have been detected"
     b = blobs[0]
-    assert b[0] == b[1] == 5, "blob should be 5 px from x and y borders"
-
+    assert b[0] == disc_center[0], f"blob should be {disc_center[0]} px from x border"
+    assert b[1] == disc_center[1], f"blob should be {disc_center[1]} px from y border"
+    
     blobs = blob_dog(
         img,
         min_sigma=1.5,
         max_sigma=5,
         sigma_ratio=1.2,
-        exclude_border=6,
+        exclude_border=exclude_border,
     )
-    msg = "zero blobs should be detected, as only blob is 5 px from border"
-    assert blobs.shape[0] == 0, msg
+    
+    if disc_center == (5, 20) and exclude_border == (4,15):
```

Comment:
I prefer to keep test's really simple and not use conditionals to test different asserts but I'm fine to keep it in this case. If you want you can use a more general test that works with different `exclude_border` values. I think it should be 

```suggestion
    if np.all(np.array(disc_center) >= np.array(exclude_border)):
```

Reply:
> I prefer to keep test's really simple and not use conditionals to test different asserts but I'm fine to keep it in this case.

Well, that is, in a way, why I thought it should be splitted to 2 functions. 

>If you want you can use a more general test that works with different exclude_border values. I think this should work

I thought about it, but I actually think that it is better to leave it like this, I think it is much easier to understand, as already from the values you can see the exact condition that it. Saves a comment I think.

Thanks